src/worker/run_query.ts

```diff
@@ -399,7 +399,33 @@ export const runQuery = async (
       sql,
       dialect,
     });
-    const queryResult = await runnable.run({rowLimit, abortSignal});
+
+    const sourceType = preparedResult.sourceExplore?.sourceStructDef?.type;
+    const sourceName = preparedResult.sourceExplore?.name;
+    // a query might not live in the same file as the source; pull location from the source
+    const sourceLocation =
+      preparedResult.sourceExplore?.sourceStructDef?.location;
+    const sourceAnnotation =
+      preparedResult.sourceExplore?.sourceStructDef?.annotation;
+    const sourceComponentInfos =
+      preparedResult.sourceExplore?.getSourceComponents();
+
+    let sourceRefWithMetadata = undefined;
+    if (sourceName && sourceLocation && sourceType) {
+      sourceRefWithMetadata = {
+        type: sourceType,
+        name: sourceName,
+        location: sourceLocation,
+        annotation: sourceAnnotation,
+        sourceComponentInfos,
+      };
+    }
```

Comment:
Nit - the long optional chaining makes it sort of hard to understand what's going on, you can get pretty much the same effect with a single conditional.

```suggestion
    let sourceRefWithMetadata = undefined;
    const {sourceExplore} = preparedResult;
    if (sourceExplore?.sourceStructDef) {
      const {name, sourceStructDef} = sourceExplore;
      // a query might not live in the same file as the source; pull location from the source
      const {type, location, annotation} = sourceStructDef;
      const sourceComponentInfos = sourceExplore.getSourceComponents();

      sourceRefWithMetadata = {
        type,
        name,
        location,
        annotation,
        sourceComponentInfos,
      };
    }
```